compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2095,6 +2119,23 @@ private BLangInvocation generateErrorCauseLanglibFunction(Location pos, BType ca
         return createLangLibInvocationNode(ERROR_CAUSE_FUNCTION_NAME, onExpr, new ArrayList<>(), causeType, pos);
     }
 
+    private BLangInvocation generateCreateRecordValueInvocation(Location pos,
```

Comment:
Can we desugar the record creation to a combination of 
1. creating a record value with a mapping constructor expression with the individual (non-rest) fields as members
2. filtering and adding the rest fields to the record value created in 1

This would be ideal.

Alternately you can create a map and call cloneWithType, but then the value has to be created twice.

One of the issues with introducing a separate method is that we would need to re-implement this record creation logic for a different runtime (e.g., nballerina),

Reply:
Creating a map and calling cloneWithType would not work since `value:cloneWithType` only supports `anydata` https://raw.githubusercontent.com/ballerina-platform/ballerina-spec/master/lang/lib/value.bal

Reply:
Created an issue to handle the approach without `langlib.internal` method.
https://github.com/ballerina-platform/ballerina-lang/issues/31087